src/main/java/io/rpg/controller/Controller.java

```diff
@@ -171,15 +173,13 @@ public void onMouseClickedEvent(MouseClickedEvent event) {
                                     .orElseThrow(() -> new RuntimeException("No object present at position " + position));
 
     double distance = playerPos.distance(objectView.getPosition());
-    if (distance < 1.5) {
-      if (object instanceof InteractiveGameObject) {
-        ((InteractiveGameObject) object).onAction();
-      }
 
-      if (object instanceof CollectibleGameObject) {
-        popupController.openTextImagePopup("Picked up an item!", objectView.getImage(), getWindowCenterX(), getWindowCenterY());
-        objectView.setVisible(false);
-        currentModel.removeGameObject(object);
+    if (distance < 1.5) {
```

Comment:
is the distance measured by squares? if yes, then 1.5 squares means i can pick up an item that is not adjacent to my square and this should not happen

Reply:
I'll look into it